src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,215 @@
+
+    private final Index index;
+    private final EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * @param index of the meeting in the filtered meeting list to edit
+     * @param editMeetingDescriptor details to edit the meeting with
+     */
+    public EditMeetingCommand(Index index, EditMeetingDescriptor editMeetingDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editMeetingDescriptor);
+
+        this.index = index;
+        this.editMeetingDescriptor = new EditMeetingDescriptor(editMeetingDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToEdit = lastShownList.get(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        if (!meetingToEdit.equals(editedMeeting) && model.hasSpecificMeeting(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        // update meeting list
+        model.setMeeting(meetingToEdit, editedMeeting);
+        model.updateFilteredMeetingList(PREDICATE_SHOW_ALL_MEETING);
+
+        // update client list
+        model.setClient(meetingToEdit.getClient(), editedMeeting.getClient());
+        model.updateFilteredClientList(PREDICATE_SHOW_ALL_CLIENTS);
+
+        return new CommandResult(String.format(MESSAGE_EDIT_MEETING_SUCCESS, editedMeeting), CommandSpecific.MEETING);
+    }
+
+    /**
+     * Creates and returns a {@code Meeting} with the details of {@code meetingToEdit}
+     * edited with {@code EditMeetingDescriptor}.
+     */
+    private static Meeting createEditedMeeting(Meeting meetingToEdit, EditMeetingDescriptor editMeetingDescriptor)
+            throws CommandException {
+        assert meetingToEdit != null;
+
+        MeetingDate updatedDate = editMeetingDescriptor.getDate().orElse(meetingToEdit.getMeetingDate());
+        Description updatedDescription = editMeetingDescriptor.getDescription().orElse(meetingToEdit.getDescription());
+        MeetingTime updatedEndTime = editMeetingDescriptor.getEndTime().orElse(meetingToEdit.getMeetingEndTime());
+        MeetingTime updatedStartTime = editMeetingDescriptor.getStartTime().orElse(meetingToEdit.getMeetingStartTime());
+        if (updatedEndTime.isBefore(updatedStartTime)) {
+            throw new CommandException(MESSAGE_END_TIME_BEFORE_START_TIME);
+        }
+        Meeting meeting = new Meeting(meetingToEdit.getClient(), updatedDescription, updatedDate,
+                updatedStartTime, updatedEndTime);
+        meeting.getClient().removeMeeting(meetingToEdit);
```

Comment:
this mutates client, which causes this transaction to not be atomic, i.e., all or nothing is not observed. so even if the transaction (editMeeting) fails, there's already side effects.